src/main/java/com/ericsson/gerrit/plugins/eiffel/configuration/EiffelPluginConfiguration.java

```diff
@@ -46,7 +64,70 @@ public EiffelPluginConfiguration(final String pluginName, final NameKey project,
             throw new ExceptionInInitializerError(String.format("Can't read %s plugin configuration for project %s: %s",
                     pluginName, project.toString(), e.getMessage()));
         }
+        
+        // Read plugin configuration
+        this.cfgEnabled = pluginConfig.getBoolean(ENABLED, false);
+        this.cfgFilter = pluginConfig.getString(FILTER);
+        this.cfgRemremGenerateURL = pluginConfig.getString(REMREM_GENERATE_URL);
+        this.cfgRemremPublishURL = pluginConfig.getString(REMREM_PUBLISH_URL);
+        this.cfgRemremUsername = pluginConfig.getString(REMREM_USERNAME);
+        this.cfgRemremPassword = pluginConfig.getString(REMREM_PASSWORD);
+        
+        //flow context is optional
+        this.cfgFlowContext = pluginConfig.getString(FLOW_CONTEXT);
+        
+        // No point to check other config parameters if plugin is disabled
+        if (!this.cfgEnabled) {
+            return;
+        }
+        
+        // Make sure that REMReM configuration is defined, otherwise we won't be able to send messages.
+        if (this.cfgRemremGenerateURL == null) {
+            throw new ExceptionInInitializerError(
+                    String.format("Can't read %s plugin configuration for project %s: REMReM Generate URL is null", pluginName,
+                            project.toString()));
+        } else if (this.cfgRemremPublishURL == null) {
+            throw new ExceptionInInitializerError(
+                    String.format("Can't read %s plugin configuration for project %s: REMReM Publish URL is null", pluginName,
+                            project.toString()));
+        } else if (this.cfgRemremUsername == null) {
+            throw new ExceptionInInitializerError(
+                    String.format("Can't read %s plugin configuration for project %s: REMReM Username is null", pluginName,
+                            project.toString()));
+        } else if (this.cfgRemremPassword == null) {
+            throw new ExceptionInInitializerError(
+                    String.format("Can't read %s plugin configuration for project %s: REMReM Password is null", pluginName,
+                            project.toString()));
+        }
```

Comment:
There is a lot of code duplication. Is it possible to create a method that checks and throws on null?

Reply:
this is needed as we need to show proper error messages on what property is missing.

Reply:
I wasn't asking to remove it I was wondering if could be simplified e.g:

```
throwOnNull(plugin, project, "REMRem Generate URL is null");
throwOnNull(plugin, project, "REMRem Publish URL is null");
...
```

Currently it is a bit confusing why we have the `else if` as I guess we want to check all values?

Reply:
I second that @m-linner-ericsson

Reply:
Modified the validations, so I only have 1 check now.

Reply:
Wasn't the other checks needed?

Reply:
I removed, because now we only intend to use REMReM Publish, and the username password are optional when you configure REMReM.